alerting/src/main/kotlin/com/amazon/opendistroforelasticsearch/alerting/alerts/AlertIndices.kt

```diff
@@ -184,6 +193,34 @@ class AlertIndices(
         }
     }
 
+    private suspend fun updateIndexMapping(index: String, alias: Boolean = false) {
+        val clusterState = clusterService.state()
+        val mapping = alertMapping()
+        var targetIndex = index
+        if (alias) {
+            targetIndex = SchemaVersionUtils.getIndexNameWithAlias(clusterState, index)
+        }
+
+        if (targetIndex == lastUpdatedHistoryIndex) {
+            return
+        }
+
+        var putMappingRequest: PutMappingRequest = PutMappingRequest(targetIndex).type(MAPPING_TYPE)
+                .source(mapping, XContentType.JSON)
+        val updateResponse: AcknowledgedResponse = client.suspendUntil { client.putMapping(putMappingRequest, it) }
+        if (updateResponse.isAcknowledged) {
+            logger.info("Index mapping of $targetIndex is updated")
+            setIndexUpdateFlag(index, targetIndex)
+        }
```

Comment:
Something should be done in an error case. At the very least add a log statement.

Reply:
Sure, will add log for error case